babel.config.js

```diff
@@ -0,0 +1,47 @@
+/* global module */
+
+module.exports = function (api) {
+  const node = {
+    ignore: [
+      'src/lib/modules/pipeline/babel-loader-overrides.js',
+      'src/lib/modules/pipeline/webpack.config.js'
+    ],
+    plugins: [
```

Comment:
I suggest using `@babel/plugin-proposal-optional-chaining`. It will make the switch to TS way easier !

Reply:
@gomesalexandre can you provide a bit more explanation, maybe given an example of how it's useful when transitioning to TS?

Reply:
This way, you don't have to check that intermediary nodes exist when trying to access a specific value. Would definitely lighten the codebase ! (Or, to be exact, not thicken it with lots of null || undefined checks when switching to statically-typed JS)

Reply:
Sounds good!  I'm working on some revisions that will be proposed in a PR to this branch. I'll include the optional chaining plugin.

Reply:
Perfect, mention me when you do I'd like to start contributing a few types already !

Reply:
@michaelsbradleyjr there is a lot of stuff that can be added/removed to this branch regarding babel configuration.
However, we will have a lot conflict due to the move src file
Maybe we focus on making this one work  as it is and make the babel config better as a second stage